packages/core/src/extractors/MetadataExtractors.ts

```diff
@@ -0,0 +1,400 @@
+      const combinedTitles = titlesCandidates.join(",");
+
+      const completion = await this.llm.complete({
+        prompt: defaultTitleCombinePromptTemplate({
+          context_str: combinedTitles,
+        }),
+      });
+
+      title = completion.text;
+    }
+
+    if (nodesToExtractTitle.length === 1) {
+      title = titlesCandidates[0];
+    }
+
+    return nodes.map((_) => ({
+      documentTitle: title.trim().replace(STRIP_REGEX, ""),
+    }));
+  }
+}
+
+/**
+ * Extract questions from a list of nodes.
+ */
+export class QuestionsAnsweredExtractor extends BaseExtractor {
+  /**
+   * LLM instance.
+   * @type {LLM}
+   */
+  llm: LLM;
+
+  /**
+   * Number of questions to generate.
+   * @type {number}
+   * @default 5
+   */
+  questions: number = 5;
+
+  /**
+   * The prompt template to use for the question extractor.
+   * @type {string}
+   * @default "{context}"
+   */
+  promptTemplate: string = defaultQuestionAnswerPromptTemplate({
+    context_str: "{context}",
+    num_questions: this.questions,
+  });
+
+  /**
+   * Wheter to use metadata for embeddings only
+   * @type {boolean}
+   * @default false
+   */
+  embeddingOnly: boolean = false;
+
+  /**
+   * Constructor for the QuestionsAnsweredExtractor class.
+   * @param {LLM} llm LLM instance.
+   * @param {number} questions Number of questions to generate.
+   * @param {string} promptTemplate The prompt template to use for the question extractor.
+   * @param {boolean} embeddingOnly Wheter to use metadata for embeddings only.
+   */
+  constructor(
+    llm: LLM,
+    questions: number = 5,
+    promptTemplate: string = defaultQuestionAnswerPromptTemplate({
+      context_str: "{context}",
+      num_questions: questions,
+    }),
+    embeddingOnly: boolean = false,
+  ) {
+    if (questions < 1) throw new Error("Questions must be greater than 0");
+
+    super();
+
+    this.llm = llm;
+    this.questions = questions;
+    this.promptTemplate = promptTemplate;
+    this.embeddingOnly = embeddingOnly;
+  }
+
+  /**
+   * Extract answered questions from a node.
+   * @param {BaseNode} node Node to extract questions from.
+   * @returns {Promise<Record<string, any>>} Questions extracted from the node.
+   */
+  async extractQuestionsFromNode(
+    node: BaseNode,
+  ): Promise<Record<string, string>> {
+    if (this.isTextNodeOnly && !(node instanceof TextNode)) {
+      return {};
+    }
+
+    const context_str = node.getContent(this.metadataMode);
+
+    const prompt = defaultQuestionAnswerPromptTemplate({
+      context_str,
+      num_questions: this.questions,
+    });
+
+    const questions = await this.llm.complete({
+      prompt,
+    });
+
+    return {
+      questionsThisExcerptCanAnswer: questions.text.replace(STRIP_REGEX, ""),
+    };
+  }
+
+  /**
+   * Extract answered questions from a list of nodes.
+   * @param {BaseNode[]} nodes Nodes to extract questions from.
+   * @returns {Promise<Record<string, any>[]>} Questions extracted from the nodes.
```

Comment:
Please type the metadata explicitly instead of Record<string, any> (for example here, you can specify {questionsThisExcerptCanAnswer: string}). Will help users better understand what the various extractors are actually doing.